### PR TITLE
Relationships

### PR DESCRIPTION
**Pull request by shella12**
In this project, I finished creating the remaining classes for the oop school library and created the associations between them.

Following changes were made:

**- Created a class Classroom with the following:**

1. @label instance variable should be initialized in the constructor.
2. Setter and getter for @label (remember about attr_accessor).

- **Create the has-many/belongs-to relationship between Classroom and Student**. The following were implemented:

1. Created the has-many side (a classroom has many students).
2. Created the belongs-to side (a student belongs to a classroom).
3. Made sure that when adding a student to a classroom it also sets the classroom for the student.
4. Made sure that when setting the classroom for a student it also adds it to the classrooms' students.

- **Created a class Book with the following:**

1. @title and @author instance variables, should be initialized in the constructor.
2. Setters and getters for instance variables (remember about attr_accessor).

- **Created a class Rental with the following:**

1. @date instance variable, should be initialized in the constructor.
2. Setter and getter for @date (remember about attr_accessor).

- **Created the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:**

1. Created the has-many side of Book and Rental (a book has many rentals).
2. Created the belongs-to side of Rental and Book (a rental belongs to a book).
3. Created the has-many side of Person and Rental (a person has many rentals).
4. Created the belongs-to side of Rental and Person (a rental belongs to a person).
5. Modified the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class Book
+  attr_reader :rents
```
Comment (FrederickMih) [inline book.rb]: - [ ] rents variable can also be modified hence let kindly use Setters and Getters(attr_accessor)

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rents = []
+  end
+
+  def add_rents(rent)
+    @rent.push(rent)
+    rent.book = self
+  end
```
Comment (FrederickMih) [inline book.rb]: - [ ] I think this method should actually create a new rental. A person on a particular date should be able to create a rental for a book. Let's kindly try doing it this way.

 ```.ruby
 def add_rents(person, date)
    Rental.new(date, self, person) # book itself should be sent as a parameter('has-many' associations)
 end
```

File: rental.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
+class Rental
+  attr_reader :student, :book, :person
```
Comment (FrederickMih) [inline rental.rb]: - [ ] Let's kindly make sure that a book and a person should be modified(`attr_accessor :date, :book, :person`) 👍

File: rental.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  attr_accessor :date
+
+  def initialize(date, student, book, person)
+    @date = date
+
+    @student = student
+    student.rents << self
+
+    @book = book
+    book.rents << self
```
Comment (FrederickMih) [inline rental.rb]: A rental belongs to a book. This is great 👍

File: person.rb [hunk 5]
```diff
@@ -20,6 +21,11 @@ def correct_name
     @name
   end
 
+  def add_rents(rent)
+    @rents.push(rent)
+    rent.person = self
+  end
```
Comment (FrederickMih) [inline person.rb]: - [ ] Let's kindly check out the comment above ⬆️ and modify this method accordingly

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi 🙋 @shella12   , 


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights 🟢 🟢 🟢 


- PR is professional ✔️ 
- Linters checks passed ✔️


## Required Changes ♻️

- [ ] **README.md** Let's kindly make sure to customize the license file. 

Capture


_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ichala — CHANGES_REQUESTED** [review]
Hi Student
Good job so far! 
👏👏👏 
You still need to work on some issues to go to the next project, but you're almost there!

the-simpsons-homer-simpson

## To Highlight 🎉

- No Linter errors  ✔️
-  All The technical requirements implemented ✔️


 ## Required Changes ♻️

- [ ] **_🛑 as mentioned by the previous reviewer kindly update your Licence information 🙏 _**






## Optional suggestions

- N/A

**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by t-yanick — APPROVED** [review]
# STATUS: MILESTONE APPROVED 👍 

Hi @shella12 , 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  All changes by previous reviewers implemented
:heavy_check_mark: Descriptive Readme file and PR

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**